Addresses GCC 14 issues https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1318
Patch is a subset of upstream commit https://github.com/micropython/micropython/commit/35f3f0a

TODO: this patch should be removed after updating to micropython 1.23.0+

diff --git a/py/nlraarch64.c b/py/nlraarch64.c
index fcc318f2dc182..d6d87ebc50db8 100644
--- a/py/nlraarch64.c
+++ b/py/nlraarch64.c
@@ -75,7 +75,7 @@ NORETURN void nlr_jump(void *val) {
         "ret                     \n"
         :
         : "r" (top)
-        :
+        : "memory"
         );

     MP_UNREACHABLE
diff --git a/py/nlrpowerpc.c b/py/nlrpowerpc.c
index 448459216b6e8..8a69fe1eeca6b 100644
--- a/py/nlrpowerpc.c
+++ b/py/nlrpowerpc.c
@@ -114,7 +114,7 @@ NORETURN void nlr_jump(void *val) {
         "blr ;"
         :
         : "r" (&top->regs)
-        :
+        : "memory"
         );

     MP_UNREACHABLE;
diff --git a/py/nlrthumb.c b/py/nlrthumb.c
index a8ffecc470335..a22c5df5b9418 100644
--- a/py/nlrthumb.c
+++ b/py/nlrthumb.c
@@ -132,7 +132,7 @@ NORETURN void nlr_jump(void *val) {
         "bx     lr                  \n" // return
         :                           // output operands
         : "r" (top)                 // input operands
-        :                           // clobbered registers
+        : "memory"                  // clobbered registers
         );

     MP_UNREACHABLE
diff --git a/py/nlrx64.c b/py/nlrx64.c
index 6b7d0262f5491..d1ad91ff7d718 100644
--- a/py/nlrx64.c
+++ b/py/nlrx64.c
@@ -123,7 +123,7 @@ NORETURN void nlr_jump(void *val) {
         "ret                        \n" // return
         :                           // output operands
         : "r" (top)                 // input operands
-        :                           // clobbered registers
+        : "memory"                  // clobbered registers
         );

     MP_UNREACHABLE
diff --git a/py/nlrx86.c b/py/nlrx86.c
index f658d41910c8d..085e30d2034a1 100644
--- a/py/nlrx86.c
+++ b/py/nlrx86.c
@@ -95,7 +95,7 @@ NORETURN void nlr_jump(void *val) {
         "ret                        \n" // return
         :                           // output operands
         : "r" (top)                 // input operands
-        :                           // clobbered registers
+        : "memory"                  // clobbered registers
         );

     MP_UNREACHABLE
diff --git a/py/nlrxtensa.c b/py/nlrxtensa.c
index abe9042af9f17..ff7af6edeef98 100644
--- a/py/nlrxtensa.c
+++ b/py/nlrxtensa.c
@@ -74,7 +74,7 @@ NORETURN void nlr_jump(void *val) {
         "ret.n                      \n" // return
         :                           // output operands
         : "r" (top)                 // input operands
-        :                           // clobbered registers
+        : "memory"                  // clobbered registers
         );

     MP_UNREACHABLE
